### PR TITLE
Allow customizing the path to the package to deploy

### PR DESCRIPTION
**Pull request by jsorge**
This PR adds the ability for the buildpack to be customized with a new `PACKAGE_DIR` config setting, which allows for pointing to a Swift package inside of the user's repository. What is enabled here is having a monorepo with other projects living side by side with a Vapor service (or services), and being able to deploy that Vapor service easily. I've been using it for a client of mine and I know of at least 1 other user of it as well and it's been a success for a couple of years now.

### CODE REVIEW

**Comment by mergesort** [comment]
This would be a big improvement for my quality of life! I've been using @jsorge's taphouseio buildpack for a couple of years now, and every time there is a change to this branch or new Swift version, I have to pester him to merge in the most recent changes into his buildpack. (Sorry and thank you so much Jared!)

The reason I use his variant is that I maintain my code in a monorepo style that looks like this.

Plinky Folder

The default buildpack has no mechanism for specifying which folder the Vapor app lives in, while Jared's workarounds do just that. Would love to see this get merged in, and happy to assist in any way I can to make that happen. 🙇🏻‍♂️

**Comment by vzsg** [comment]
If you could align whitespaces to the existing ones (two spaces instead of one tab) in bin/detect and bin/compile, I'm okay with merging this in and pushing it out to the buildpack registry. Assuming that it does not break the common projects where the project lives in the root.

File: bin/compile [hunk 1]
```diff
@@ -7,20 +7,35 @@ set -o pipefail
...
 
+VAR_PATH="$ENV_DIR/PACKAGE_DIR"
+
+if [[ -f "$VAR_PATH" ]]; then
+	package=$(cat "$VAR_PATH")
+	BUILD_DIR="$APP_DIR/$package"
+	echo "Using package directory at $package"
+else
+	BUILD_DIR="$APP_DIR"
+	echo "No PACKAGE_DIR config variable is set. Using the repo root."
```
Comment (vzsg) [inline bin/compile:22]:
```suggestion
  package=$(cat "$VAR_PATH")
  BUILD_DIR="$APP_DIR/$package"
  echo "Using package directory at $package"
else
  BUILD_DIR="$APP_DIR"
  echo "No PACKAGE_DIR config variable is set. Using the repo root."
```

File: bin/compile [hunk 2]
```diff
@@ -7,20 +7,35 @@ set -o pipefail
...
+fi
+
 SWIFT_VERSION="6.0.2"
+CLANG_VERSION=7.0.1
```
Comment (vzsg) [inline bin/compile:26]:
```suggestion
```

This is no longer necessary.

File: bin/compile [hunk 3]
```diff
@@ -7,20 +7,35 @@ set -o pipefail
...
 source "$BIN_DIR/utils"
 
 if [ -f "$BUILD_DIR/.swift-version" ]; then
-  SWIFT_VERSION=`cat $BUILD_DIR/.swift-version | tr -d '[[:space:]]'`
-  puts-step "Using Swift $SWIFT_VERSION (from .swift-version file)"
+	SWIFT_VERSION=$(cat $BUILD_DIR/.swift-version | tr -d '[[:space:]]')
+	puts-step "Using Swift $SWIFT_VERSION (from package .swift-version file)"
+elif [ -f "$APP_DIR/.swift-version" ]; then
+	SWIFT_VERSION=$(cat $APP_DIR/.swift-version | tr -d '[[:space:]]')
+	puts-step "Using Swift $SWIFT_VERSION (from top-level .swift-version file)"
```
Comment (vzsg) [inline bin/compile:38]:
```suggestion
  SWIFT_VERSION=$(cat $BUILD_DIR/.swift-version | tr -d '[[:space:]]')
  puts-step "Using Swift $SWIFT_VERSION (from package .swift-version file)"
elif [ -f "$APP_DIR/.swift-version" ]; then
  SWIFT_VERSION=$(cat $APP_DIR/.swift-version | tr -d '[[:space:]]')
  puts-step "Using Swift $SWIFT_VERSION (from top-level .swift-version file)"
```

File: bin/compile [hunk 4]
```diff
@@ -58,8 +73,13 @@ if [[ -z "$RUN_TESTS" ]]; then
 fi
 
 # Setup application environment
-mkdir -p $BUILD_DIR/.profile.d
+set-env() {
+	PROFILE_PATH="$APP_DIR/.profile.d/swift.sh"
+	echo "export $1=$2" >>$PROFILE_PATH
```
Comment (vzsg) [inline bin/compile:78]:
```suggestion
  PROFILE_PATH="$APP_DIR/.profile.d/swift.sh"
  echo "export $1=$2" >>$PROFILE_PATH
```

File: bin/detect [hunk 5]
```diff
@@ -1,8 +1,13 @@
...
+BUILD_DIR=$1
+
+MANIFEST='Package.swift'
+RESULTS=$(find "$BUILD_DIR" -name $MANIFEST)
+
+if [[ "$RESULTS" == *"$MANIFEST"* ]]; then
+	echo "Swift" && exit 0
 else
-  echo "no" && exit 1
+	echo "no" && exit 1
```
Comment (vzsg) [inline bin/detect:12]:
```suggestion
  echo "no" && exit 1
```

File: bin/detect [hunk 6]
```diff
@@ -1,8 +1,13 @@
...
 
-if [[ -f $1/Package.swift ]]; then
-  echo "Swift" && exit 0
+BUILD_DIR=$1
+
+MANIFEST='Package.swift'
+RESULTS=$(find "$BUILD_DIR" -name $MANIFEST)
+
+if [[ "$RESULTS" == *"$MANIFEST"* ]]; then
+	echo "Swift" && exit 0
```
Comment (vzsg) [inline bin/detect:10]:
```suggestion
  echo "Swift" && exit 0
```

**Review by vzsg — CHANGES_REQUESTED** [review]

**Comment by jsorge** [comment]
This won't break any existing projects since the new config property is entirely optional. Thanks for the feedback!

**Pull request merged**